src/hd/hd.c

```diff
@@ -5675,6 +5676,33 @@ void read_udevinfo(hd_data_t *hd_data)
 
   s = free_mem(s);
 
+  /*
+   * It sometimes happens that udev generates the same link for different
+   * kernel devices. To catch this we check here that udev device symlinks
+   * actually point to the kernel device name.
+   *
+   * If it does not match the link is replaced by the kernel device name.
+   */
+  for(ui = hd_data->udevinfo; ui; ui = ui->next) {
+    if(!ui->name || stat(ui->name, &sbuf)) continue;
```

Comment:
Since you don't need `sbuf` anymore after this, why not simply use

    -e ui->name

?

Reply:
This is C, mind.

Reply:
...or, better yet,

    -f ui->name

Reply:
Duh - I was in Perl land, but this is C, so it would be

    access(...)

Reply:
Basically the same; I just prefer stat().